crates/wasi-common/src/snapshots/preview_1.rs

```diff
@@ -1554,7 +1554,13 @@ fn dirent_bytes(dirent: types::Dirent) -> Vec<u8> {
     let mut bytes = Vec::with_capacity(size);
     bytes.resize(size, 0);
     let ptr = bytes.as_mut_ptr() as *mut types::Dirent;
-    unsafe { ptr.write_unaligned(dirent) };
+    let guest_dirent = types::Dirent {
+        d_ino: dirent.d_ino.to_le(),
+        d_namlen: dirent.d_namlen.to_le(),
+        d_type: dirent.d_type, // endian-invariant
```

Comment:
Could this assert that the size of this field is 1 so the endianness is known to not matter?

Reply:
Like this?   Patch updated.